### PR TITLE
feat(sdk): Add ability to compile If statements and ternary expressions to graph

### PR DESCRIPTION
**Pull request by mrzzy**
## Why we need this PR/What is changed.
Follow up of PR #7.

Improves user experience building the computational graph:
- Allows users to write normal if statements to create switch nodes
   ```python
    y, m, n = "str1", "str2", "str3"
    if True:
        x = y
        z = 1
    elif False:
        x = m
        z = 2
    else:
        x = n
        z = 3
   ```
   is now transformed by the Graph Compiler to:
   ```python
    (y, m, n) = ('str1', 'str2', 'str3')
    if True:

        def __if_block(m, n, y):
            x = y
            z = 1
            return (x, z)

        def __else_block(m, n, y):
            if True:

                def __if_block(m, n):
                    x = m
                    z = 2
                    return (x, z)

                def __else_block(m, n):
                    x = n
                    z = 3
                    return (x, z)
                __if_outputs = __if_block(m=m, n=n)
                __else_outputs = __else_block(m=m, n=n)
                (x, z) = [g.switch(condition=False, true=if_out, false=else_out) for (if_out, else_out) in zip(__if_outputs, __else_outputs)]
            return (x, z)
        __if_outputs = __if_block(m=m, n=n, y=y)
        __else_outputs = __else_block(m=m, n=n, y=y)
        (x, z) = [g.switch(condition=True, true=if_out, false=else_out) for (if_out, else_out) in zip(__if_outputs, __else_outputs)]
   ```
- Allows users to write ternary expressions to create switch nodes
  ```python
  x = y if True else b
  ```
  is now transformed by the Graph Compiler to:
  ```python
   x = g.switch(condition=True, true=y, false=b)
  ````

Fixes bugs:
- Fix typo `__name___` should be `__name__`
- Fix issue where  GraphNodes are getting added to GraphComponent's inputs instead of expected `Node.Retrieve` when using Augment assign (ie `+=`)

Improves test coverage, particularly on the generated computation graph.

### CODE REVIEW

**Comment by mrzzy** [comment]
Blocked until PR #7 is merged.

File: sdk/bento/graph/ecs.py
Comment (joeltio) [inline sdk/bento/graph/ecs.py:244]: I don't think the entity ID should be deterministic. If I have two entities with the same components and same values in their components, it doesn't mean that there is only one entity.

File: sdk/bento/graph/ecs.py
Comment (joeltio) [inline sdk/bento/graph/ecs.py:263]: `type(self).__name__`

**Review by joeltio — CHANGES_REQUESTED** [review]

File: sdk/bento/graph/ecs.py
Comment (mrzzy) [reply sdk/bento/graph/ecs.py:244]: This will be eventually replaced with Entity Id from set by the Engine. Right now I need it be deterministic in order to write tests.

File: sdk/bento/graph/ecs.py
Comment (mrzzy) [reply sdk/bento/graph/ecs.py]: Updated.

**Review by joeltio — APPROVED** [review]
LGTM :+1:

**Pull request merged**
